akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,23 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case ActorSelectionMessage(msg, elements, _) ⇒ {
+      elements.foreach {
+        case SelectChildName(name) ⇒ getChild(name).tell(msg, sender)
+        case pattern @ SelectChildPattern(_) ⇒ {
+          import scala.collection.JavaConverters.mapAsScalaConcurrentMapConverter
+          children.asScala filter {
+            case (name, child) ⇒ pattern.pattern.matcher(name).matches
+          } foreach {
+            case (name, child) ⇒ child.tell(msg, sender)
+          }
+        }
+        case SelectParent ⇒ getParent.tell(msg, sender)
```

Comment:
this not correct, e.g. `"/temp/../temp/$a"

it should only send `msg` to parent if sel.elements.drop(1).isEmpty otherwise it should send `ActorSelectionMessage(sel.elements.drop(1), msg, sender)`